src/dialects/db2/query-generator.js

```diff
@@ -181,8 +181,20 @@ export class Db2QueryGenerator extends AbstractQueryGenerator {
     });
   }
 
-  showTablesQuery() {
-    return 'SELECT TABNAME AS "tableName", TRIM(TABSCHEMA) AS "tableSchema" FROM SYSCAT.TABLES WHERE TABSCHEMA = USER AND TYPE = \'T\' ORDER BY TABSCHEMA, TABNAME';
+  showTablesQuery(_database, options) {
```

Comment:
Same here, unify `database` (mariadb, but is actually schema) with `options.schema` into either of these two signatures:

```ts
showTablesQuery(schema?: string): string;
// or
showTablesQuery(options?: { schema?: string }): string;
```

Reply:
this may benefit from the dialect `defaultSchema` call.  I imagine that could be a future refactor commit/PR.